manim/config.py

```diff
@@ -1,283 +1,406 @@
             logger.error(err)
             sys.exit(2)
 
-    # If rendering a transparent image/move, make sure the
-    # scene has a background opacity of 0
-    if args.transparent:
-        camera_config["background_opacity"] = 0
-
-    return camera_config
+    # As before, make FILE_WRITER_CONFIG by pulling form the overall
+    config['FILE_WRITER_CONFIG'] = {key: config[key] for key in [
+        "WRITE_TO_MOVIE", "SAVE_LAST_FRAME", "SAVE_PNGS",
+        "SAVE_AS_GIF", "PNG_MODE", "MOVIE_FILE_EXTENSION",
+        "OUTPUT_FILE", "INPUT_FILE"]}
 
+    return config
 
-def initialize_directories(config):
-    dir_config = {}
-    dir_config["media_dir"] = config["media_dir"] or dirs.MEDIA_DIR
-    dir_config["video_dir"] = config["video_dir"] or dirs.VIDEO_DIR
 
-    if not (config["video_dir"] and config["tex_dir"]):
-        if config["media_dir"]:
-            if not os.path.isdir(dir_config["media_dir"]):
-                os.makedirs(dir_config["media_dir"])
-        if not os.path.isdir(dir_config["media_dir"]):
-            dir_config["media_dir"] = "./media"
+def _init_dirs(config):
+    if not (config["VIDEO_DIR"] and config["TEX_DIR"]):
+        if config["MEDIA_DIR"]:
+            if not os.path.isdir(config["MEDIA_DIR"]):
+                os.makedirs(config["MEDIA_DIR"])
+        if not os.path.isdir(config["MEDIA_DIR"]):
+            config["MEDIA_DIR"] = "./media"
         else:
             print(
-                f"Media will be written to {dir_config['media_dir'] + os.sep}. You can change "
-                "this behavior with the --media_dir flag, or by adjusting dirs.py.,"
+                f"Media will be written to {config['media_dir'] + os.sep}. You can change "
+                "this behavior with the --media_dir flag, or by adjusting manim.cfg."
             )
     else:
-        if config["media_dir"]:
+        if config["MEDIA_DIR"]:
             print(
                 "Ignoring --media_dir, since both --tex_dir and --video_dir were passed."
             )
 
-    dir_config["tex_dir"] = (config["tex_dir"]
-                             or dirs.TEX_DIR
-                             or os.path.join(dir_config["media_dir"], "Tex"))
-    dir_config["text_dir"] = (config["text_dir"]
-                              or dirs.TEXT_DIR
-                              or os.path.join(dir_config["media_dir"], "texts"))
+    # Make sure all folders exist
+    for folder in [config["VIDEO_DIR"], config["TEX_DIR"], config["TEXT_DIR"]]:
+        if not os.path.exists(folder):
+            os.makedirs(folder)
 
-    if not config["video_dir"] or dirs.VIDEO_DIR:
-        dir_config["video_dir"] = os.path.join(dir_config["media_dir"], "videos")
 
-    for folder in [dir_config["video_dir"], dir_config["tex_dir"], dir_config["text_dir"]]:
-        if folder != "" and not os.path.exists(folder):
-            os.makedirs(folder)
+config_filename = 'manim.cfg'
+
+foobar =  os.path.join(os.path.dirname(__file__), config_filename)
```

Comment:
Not sure if I need [`__path__`](https://docs.python.org/3/tutorial/modules.html#packages-in-multiple-directories) here. In fact, when trying to use it, I get a `NameError` (not defined).

Reply:
Yeah, I can't find a good way to get the package name either now that I've tried it. I guess this will have to depend on the locations of both `config.py` and `manim.cfg`.